code/game/objects/structures/wasteplants.dm

```diff
@@ -0,0 +1,58 @@
+/obj/structure/flora/wasteplant
+	name = "wasteland plant"
+	desc = "It's a wasteland plant."
+	icon = 'icons/obj/flora/plants.dmi'
+	anchored = 1
+	density = 0
+	var/has_plod = 1
+	var/obj/item/reagent_containers/food/snacks/grown/produce = null
+
+/obj/structure/flora/wasteplant/attack_hand(mob/user)
+	if(has_plod)
+		user.put_in_hands(new produce)
+		user << "<span class='notice'>You take [produce] from [src].</span>"
+		has_plod = !has_plod
+		regrow()
+	update_icon()
+
+/obj/structure/flora/wasteplant/proc/regrow()
+	if(!has_plod)
+		spawn(5000)
```

Comment:
make sure `has_plod = !has_plod` and `update_icon()` is on the same level as `sleep(...)`
```suggestion
		sleep(5000)
```